test/test_parse_es.py

```diff
@@ -84,6 +86,134 @@ def test_extract_number_es(self):
+            datetime(1998, 6, 11))
+        self.assertEqual(extract_datetime(
+            "11 jul", lang='es', anchorDate=datetime(1998, 7, 1))[0],
+            datetime(1998, 7, 11))
+        self.assertEqual(extract_datetime(
+            "11 ago", lang='es', anchorDate=datetime(1998, 8, 1))[0],
+            datetime(1998, 8, 11))
+        self.assertEqual(extract_datetime(
+            "11 sep", lang='es', anchorDate=datetime(1998, 9, 1))[0],
+            datetime(1998, 9, 11))
+
+        # It's also failing on years
+        self.assertEqual(extract_datetime(
+            "11 ago 1998", lang='es')[0], datetime(1998, 8, 11))
+
+    def test_extract_datetime_relative(self):
+        self.assertEqual(extract_datetime("esta noche", anchorDate=datetime(
+            1998, 1, 1), lang='es'),
+            [datetime(1998, 1, 1, 21, 0, 0), 'esta'])
+        self.assertEqual(extract_datetime(
+            "ayer noche", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 31, 21))
+        self.assertEqual(extract_datetime(
+            "el noche anteayer", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 30, 21))
+        self.assertEqual(extract_datetime(
+            "el noche ante ante ayer", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 29, 21))
+        self.assertEqual(extract_datetime(
+            "mañana en la mañana", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1998, 1, 2, 8))
+        self.assertEqual(extract_datetime(
+            "ayer en la tarde", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1997, 12, 31, 15))
+
+        self.assertEqual(extract_datetime(
+            "que año es", anchorDate=datetime(1998, 1, 1),
+            lang='es')[0], datetime(1998, 1, 1))
+
+        self.assertEqual(extract_datetime("hoy 2 a.m.", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 2))
+        self.assertEqual(extract_datetime("hoy 2 am", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 2))
+        self.assertEqual(extract_datetime("hoy 2 p.m.", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 14))
+        self.assertEqual(extract_datetime("hoy 2 pm", lang='es',
+                                          anchorDate=datetime(1998, 1, 1))[0],
+                         datetime(1998, 1, 1, 14))
+
+    @unittest.skip("These phrases are not parsing correctly.")
+    # parses as "morning" and returns 8:00 on anchorDate
+    def test_extract_datetime_relative_failing(self):
+        self.assertEqual(extract_datetime(
+            "mañana", anchorDate=datetime(1998, 1, 1), lang='es')[0],
+            datetime(1998, 1, 2))
```

Comment:
When I tried to create parser_es.py I tried to differentiate "morning" from "tomorrow" by parsing the former word. In spanish "la mañana" is _always_ the time frame before midday, while the day after today is simply "mañana". That's the only way we could differentiate those two terms: if "mañana" is preceded by "la" is unmistakably "morning" and without is unmistakably "tomorrow", so even "mañana por la mañana" can be parsed correctly.

Reply:
I thought this would be a trivial addition, add another condition there, but it turns out there are a number of places in the parser that require some tweaking to facilitate this. If one of you guys wants to take a stab, please do!

Reply:
relevant to #64

@adocampo apart from "esta mañana," are there any other exceptions that we'll need to account for?

Reply:
When you say other exceptions you mean all combinations of  "_whatever_ mañana" meaning "morning" don't you? For example
"esta mañana" means "this morning"
"hoy por/en la mañana" means "today in the morning"
"mañana por/en la mañana" means "tomorrow morning"

So, no. I think generally we can say "la mañana" for the time frame between midnight and noon with the sole exception of "esta mañana" which means "this morning".

Good point!